js/runner_test.go

```diff
@@ -2745,3 +2745,35 @@ exports.default = () => {
 		})
 	}
 }
+
+func TestArchivingAnArchiveWorks(t *testing.T) {
```

Comment:
This test passes on `master` as well.

Doesn't it need to assert that the options _weren't_ updated after the second `NewFromArchive()` call?

Reply:
It is passing on master as it isn't fixing anything it is just a test that I broke on the original writing of this.  If this commit was fixing somethign it wouldn't be a refactor ;) 

And the point was that I broke the whole *making* of the archive from the archive. Which is also what the names suggests

Reply:
OK, makes sense.